### DISCUSSION TITLE
MongoDB NoSql databases

### DISCUSSION THREAD

**Discussion by MaxiDutra** [pocketbase/pocketbase#58] [Q&A] [answered]
Greatings from Uruguay! i really like this project i think that expand to a non relational database would be a really cool feature for the next updates. 

Keep going amazing work  :)

**Comment by ganigeorgiev** [comment, marked as answer]
Supporting a non relation database will require a major rewrite, since PocketBase heavily relies on the SQL syntax for almost everything (even the collection API rules are translated to SQL WHERE conditions).

Support for other SQL databases (MySQL, PostgreSQL) could be added in the future, bit it's a low priority.
You could follow the discussion here - https://github.com/pocketbase/pocketbase/discussions/14

**Reply by greymont** [reply]
I hope you consider Firebird if thinking about adding other SQL database support. It is also a very mature FOSS minimal-installation-requirement rdbms.

**Reply by zhenruyan** [reply]
and  leveldb

**Comment by pchasco** [comment]
Just heard about this project and very interested. SQLite has rich support for JSON, including in-place update of objects using property paths. If you require a document database over relational, you may be able to leverage those JSON features of SQLite, providing they are available to you via PocketBase's API.

For many developers, "NoSQL" just means "no schema," which you can easily achieve with TEXT columns and JSON functions in SQLite. A misconception that developers have of "NoSQL" databases is that they were invented to solve a design problem that a relational database cannot solve. This is not the case. You can easily use a relational database as a document database. Modern "NoSQL" databases were invented to solve a scaling problem, because a relational database does not scale well horizontally. And, as PocketBase is not intended to scale horizontally, it is a non-issue.
